src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryBlock.cs

```diff
@@ -13,14 +13,16 @@ namespace System.Reflection.Internal
     [DebuggerDisplay("{GetDebuggerDisplay(),nq}")]
     internal unsafe struct MemoryBlock
     {
-        internal readonly byte* Pointer;
-        internal readonly int Length;
+        internal readonly byte* StartPointer;
+        internal readonly byte* EndPointer;
+
+        internal int Length => (int)(EndPointer - StartPointer);
 
         internal MemoryBlock(byte* buffer, int length)
         {
             Debug.Assert(length >= 0 && (buffer != null || length == 0));
```

Comment:
GetPointerChecked below does a check to see if StartPointer+offset would overflow and throws if it does.  Should we do the same here with buffer+length, even if just as an assert?